tests/jax/test_distributed_softmax.py

```diff
@@ -81,3 +95,51 @@ def ref_func(x, mask):
                         metric_bwd_dtype=dtype,
                         in_shardings=(x_pspec, mask_pspec),
                         out_shardings=(None, (x_pspec,)))
+
+    @pytest.mark.parametrize('device_count,mesh_shape,mesh_axes,mesh_resource', generate_configs())
+    @pytest.mark.parametrize('data_shape', [[32, 12, 128, 128]])
+    @pytest.mark.parametrize(
+        'softmax_type',
+        [SoftmaxType.SCALED, SoftmaxType.SCALED_MASKED, SoftmaxType.SCALED_UPPER_TRIANG_MASKED])
+    def test_softmax_bad_sharding(self, device_count, mesh_shape, mesh_axes, mesh_resource, data_shape,
```

Comment:
Can't we just reuse the `test_softmax_sharding` with `@pytest.mark.parametrize('bad_sharding', [True, False])` and only check the warning messages when `bad_sharding == True`?